### PR TITLE
Add TPM local attestation functionality, some caps and fixes nits

### PR DESCRIPTION
**Pull request by zaolin**
Signed-off-by: Philipp Deppenwiese <zaolin@das-labor.org>

### CODE REVIEW

File: pkg/tss/keys.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	if err != nil {
+		return nil, err
+	}
+	sOpt := tpm2tools.SealCurrent{
+		PCRSelection: tpm2.PCRSelection{
+			Hash: tpm2.AlgSHA256,
+			PCRs: pcrs,
+		},
+	}
+	sealed, err := key.Seal(data, sOpt)
```
Comment (hugelgupf) [inline pkg/tss/keys.go:122]: Just `return key.Seal...`?

Applies to a whole bunch of other funcs here too

File: pkg/tss/keys.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	}
+	srkTemplate := tpm2.Public{
+		Type:          tpm2.AlgECC,
+		NameAlg:       tpm2.AlgSHA256,
+		Attributes:    tpm2.FlagStorageDefault,
+		ECCParameters: defaultECCParams(),
+		AuthPolicy:    srkAuth,
+	}
+
+	key, err := tpm2tools.NewCachedKey(rwc, tpm2.HandleOwner, srkTemplate, tpm2tools.SRKECCReservedHandle)
```
Comment (hugelgupf) [inline pkg/tss/keys.go:61]: Just return this

File: pkg/tss/keys.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+func unseal12(rwc io.ReadWriteCloser, srkPW string, sealed []byte) ([]byte, error) {
+	var srkAuth [20]byte
+
+	if srkPW != "" {
+		srkAuth = sha1.Sum([]byte(srkPW))
+	}
+	unsealed, err := tpm.Unseal(rwc, sealed, srkAuth[:])
+	if err != nil {
+		return nil, fmt.Errorf("couldn't seal the data", err)
```
Comment (hugelgupf) [inline pkg/tss/keys.go:105]: Need a format specifier. "could not seal data: %v"

Here and elsewhere

File: pkg/tss/keys.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+package tss
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"crypto/sha256"
+	"fmt"
+	"io"
+
+	"github.com/google/go-tpm-tools/proto"
```
Comment (hugelgupf) [inline pkg/tss/keys.go:14]: This proto will add quite some binary size. Are you sure you want it?

Comment (zaolin) [reply pkg/tss/keys.go:13]: That's fine

**Comment by rminnich** [comment]
where's this going? will it remain a WIP?

File: pkg/tss/keys.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+		PCRSelection: tpm2.PCRSelection{
+			Hash: tpm2.AlgSHA256,
+			PCRs: pcrs,
+		},
+	}
+
+	return key.Seal(data, sOpt)
+}
+
+func unseal20(rwc io.ReadWriteCloser, srkPW string, pcrs []int, sealed *proto.SealedBytes) ([]byte, error) {
```
Comment (hugelgupf) [inline pkg/tss/keys.go:112]: i don't see these seal/reseal/unseal funcs being used anywhere?

Comment (zaolin) [reply pkg/tss/keys.go:112]: Not yet. I will do follow-ups which will implement heads in golang as booter

**Comment by hugelgupf** [comment]
2nd time i've seen the integration tests time out. retrying.

**Pull request merged**